src/compiler/scala/tools/nsc/backend/jvm/ClassfileWriters.scala

```diff
@@ -172,7 +196,8 @@ abstract class ClassfileWriters {
     val jarWriter: JarOutputStream = {
       val manifest = new Manifest()
       mainClass foreach { c => manifest.getMainAttributes.put(Name.MAIN_CLASS, c) }
-      val jar = new JarOutputStream(new BufferedOutputStream(new FileOutputStream(file.file), 64000), manifest)
+      plugins foreach (_.augmentManifest(manifest))
```

Comment:
Could we just leave manifest augmentation to the custom `JarFactory`? That would mean one fewer extension point.

Reply:
but then the JarFactory would need to know the Global, or the Plugins?

I tend to think that the Manifest should be written anyway, even to a directory. The Manifest here is a little confused for multi-destination output ( which I am not sure that anyone uses anyway), but outside the scope of this change